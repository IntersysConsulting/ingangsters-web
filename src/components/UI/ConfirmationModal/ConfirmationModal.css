.confirmationModalHeader {
  background-color: var(--green);
  color: white;
}

.confirmationNegativeButton,
.confirmationAffirmativeButton {
  background-color: none;
}

.confirmationAffirmativeButton {
  border: 1px solid var(--red);
  color: var(--red);
}

.confirmationNegativeButton {
  border: 1px solid var(--light-blue);
  color: var(--light-blue);
}
