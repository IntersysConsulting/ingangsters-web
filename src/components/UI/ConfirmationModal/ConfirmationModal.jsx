import React from "react";
import Modal from "react-bootstrap/Modal";
import Button from "react-bootstrap/Button";
import "./ConfirmationModal.css";
const ConfirmationModal = ({
  title,
  message,
  negativeText = "No",
  negativeAction,
  affirmativeText = "Yes",
  affirmativeAction,
  show,
  closeAction = () => {}
}) => {
  return (
    <Modal show={show} onHide={closeAction}>
      <Modal.Header closeButton className="confirmationModalHeader">
        <Modal.Title>{title}</Modal.Title>
      </Modal.Header>
      <Modal.Body>{message}</Modal.Body>
      <Modal.Footer className="confirmationModalFooter">
        <Button
          variant="light"
          className="confirmationNegativeButton"
          onClick={negativeAction}
        >
          {negativeText}
        </Button>
        <Button
          variant="light"
          className="confirmationAffirmativeButton"
          onClick={affirmativeAction}
        >
          {affirmativeText}
        </Button>
      </Modal.Footer>
    </Modal>
  );
};

export default ConfirmationModal;
