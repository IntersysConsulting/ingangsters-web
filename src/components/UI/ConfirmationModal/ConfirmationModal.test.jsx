import React from "react";
import { shallow } from "enzyme";
import ConfirmationModal from "./ConfirmationModal";
import sinon from "sinon";

describe("Confirmation Modal", () => {
  it("renders properly when show=true", () => {
    const shallowRender = shallow(
      <ConfirmationModal
        title="Test Render"
        message="Test Message"
        negativeText="Decline"
        negativeAction={() => {}}
        affirmativeText="Accept"
        affirmativeAction={() => {}}
        show={true}
      />
    );
    expect(shallowRender).toMatchSnapshot();
  });

  it("calls close function, negative and affirmative actions once each", () => {
    const affirmativeSpy = sinon.spy();
    const negativeSpy = sinon.spy();

    const wrapper = shallow(
      <ConfirmationModal
        title="Test Render"
        message="Test Message"
        negativeText="Decline"
        negativeAction={negativeSpy}
        affirmativeText="Accept"
        affirmativeAction={affirmativeSpy}
        show={true}
      />
    );
    wrapper
      .find(".confirmationAffirmativeButton")
      .first()
      .simulate("click");
    wrapper
      .find(".confirmationNegativeButton")
      .first()
      .simulate("click");
    expect(affirmativeSpy.calledOnce).toBe(true);
    expect(negativeSpy.calledOnce).toBe(true);
  });
});
