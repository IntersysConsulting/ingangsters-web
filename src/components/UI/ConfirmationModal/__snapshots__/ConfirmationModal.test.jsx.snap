// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation Modal renders properly when show=true 1`] = `
<Bootstrap(Modal)
  onHide={[Function]}
  show={true}
>
  <ModalHeader
    className="confirmationModalHeader"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle>
      Test Render
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    Test Message
  </ModalBody>
  <ModalFooter
    className="confirmationModalFooter"
  >
    <Button
      active={false}
      className="confirmationNegativeButton"
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="light"
    >
      Decline
    </Button>
    <Button
      active={false}
      className="confirmationAffirmativeButton"
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="light"
    >
      Accept
    </Button>
  </ModalFooter>
</Bootstrap(Modal)>
`;
